server/admin/prompt.go

```diff
@@ -12,26 +13,53 @@ import (
 	"golang.org/x/crypto/ssh/terminal"
 )
 
+type passwordReadResult struct {
+	password []byte
+	err      error
+}
+
```

Comment:
I don't think you need a named type here, you could do it inline. You also have a variable with the same name later on.

Reply:
I could do that... but it'll just be a bit of work trying to read the returned password and err later in the switch case that waits on the done channel. I should rename the variable with the same name as the type to prevent ambiguity and confusion.

Reply:
I think the named type does help, but it could be defined inside `PasswordPrompt`.